connectors/preflight_check.py

```diff
@@ -0,0 +1,71 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+
+from connectors.byoc import CONNECTORS_INDEX, JOBS_INDEX
+from connectors.es import ESClient
+from connectors.logger import logger
+from connectors.utils import CancellableSleeps
+
+
+class PreflightCheck:
```

Comment:
I don't think it's useful to have the preflight check implemented as a service because it's a single synchronous check you do before you start all work.

I think the pattern we are looking for in `_start_service` is:


```

def _shutdown_service(sig, services):
    for service in services:
        service.shutdown()

async _start_service():
     preflight_check()
     
     # now we can prepare services
     services = [...]

     # and prepare a shutdown hook
     for sig in (signal.SIGINT, signal.SIGTERM):
        loop.add_signal_handler(sig, functools.partial(_shutdown_service, sig, services))

    runners = [service.run() for service in services]
    await asyncio.gather(*services)
     
```

Reply:
As we discussed today, we want to make it a service because we want preflight check to be able to shutdown gracefully.